### PR TITLE
[Keyboard] Refactor JJ50 keyboard mapping to current standard

### PR DESCRIPTION
**Pull request by GreatWizard**
## Description

* [Doc] Update main README (https://github.com/GreatWizard/qmk_firmware/blob/update_default_jj50/keyboards/jj50/README.md)
* [Doc] Add README about the default keymaps (https://github.com/GreatWizard/qmk_firmware/blob/update_default_jj50/keyboards/jj50/keymaps/default/README.md)
- [Feature] Add `Dvroak`, `Colemak` as base layers
* [Feature] Add an `Adjust` layer to change the base layer
* [Enhancement] Update the mappings in all the layers to follow the default 5x12 ortholinear preonic
* [Enhancement] Update `Fn` mapping to have all rgblight and backlight features

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/jj50/keymaps/default/keymap.c [hunk 1]
```diff
@@ -19,117 +20,200 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
-#define ______ KC_TRNS
-#define _DEFLT 0
-#define _RAISE 1
-#define _LOWER 2
-#define _FN 3
-
-enum custom_keycodes {
-    P_MACRO = SAFE_RANGE
+enum layers {
+  _QWERTY,
```
Comment (fauxpark) [inline keyboards/jj50/keymaps/default/keymap.c:24]: Our current coding conventions specify 4 spaces: https://docs.qmk.fm/#/coding_conventions_c

File: keyboards/jj50/keymaps/default/keymap.c [hunk 2]
```diff
@@ -19,117 +20,200 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+  * |------+------+------+------+------+------+------+------+------+------+------+------|
+  * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+  * `-----------------------------------------------------------------------------------'
+  */
+  [_LOWER] = LAYOUT( \
+    KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC, \
+    KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,  \
+    KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE, \
+    _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______, \
+    _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY \
```
Comment (fauxpark) [inline keyboards/jj50/keymaps/default/keymap.c:126]:
```suggestion
  [_LOWER] = LAYOUT(
    KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC,
    KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,
    KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE,
    _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______,
    _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

File: keyboards/jj50/keymaps/default/keymap.c [hunk 3]
```diff
@@ -19,117 +20,200 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+  * |------+------+------+------+------+------+------+------+------+------+------+------|
+  * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+  * `-----------------------------------------------------------------------------------'
+  */
+  [_RAISE] = LAYOUT( \
+    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,  \
+    KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS, \
+    _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______, \
+    _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY  \
```
Comment (fauxpark) [inline keyboards/jj50/keymaps/default/keymap.c:147]:
```suggestion
  [_RAISE] = LAYOUT(
    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
    KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
    _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______,
    _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

File: keyboards/jj50/keymaps/default/keymap.c [hunk 4]
```diff
@@ -19,117 +20,200 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+        set_single_persistent_default_layer(_COLEMAK);
+      }
+      return false;
+      break;
+    case DVORAK:
+      if (record->event.pressed) {
+          set_single_persistent_default_layer(_DVORAK);
+      }
+      return false;
+      break;
```
Comment (fauxpark) [inline keyboards/jj50/keymaps/default/keymap.c:216]:
```suggestion
    case COLEMAK:
      if (record->event.pressed) {
        set_single_persistent_default_layer(_COLEMAK);
      }
      return false;
    case DVORAK:
      if (record->event.pressed) {
          set_single_persistent_default_layer(_DVORAK);
      }
      return false;
```

**Comment by fauxpark** [comment]
With the changes to the `archetype` keymap this PR must go through breaking changes.

**Comment by GreatWizard** [comment]
I can't delete the rules.mk for the `archetype` keymap, there is 2 more rules then the keyboard-level one:
```
TAP_DANCE_ENABLE = yes
AUTO_SHIFT_ENABLE = yes
```

**Comment by fauxpark** [comment]
Ah, well you can delete everything else then.

**Comment by GreatWizard** [comment]
The CI is having a hard time with xvzver, returning `error: NKRO not supported with this protocol`
And I have no idea why, NKRO is disabled in JJ50 rules

**Comment by fauxpark** [comment]
`layouts\community\ortho_5x12\xyverz\rules.mk` is meant to check whether `PROTOCOL` is `VUSB`, which is set in `quantum/mcu_selection.mk` however the rules.mk is actually evaluated first. You can safely ignore that for now.

**Comment by GreatWizard** [comment]
My latest commit adapt a bit the workflow... I'm not sure if there is any blockers by changing the order ?

Can we talk somewhere else in order to chat about the subject? (discord irc slack...)

**Comment by fauxpark** [comment]
Interesting that that seems to work, considering mcu_selection.mk first needs to know what `MCU` is, which is defined in rules.mk. I suppose it must have to do with when make evaluates stuff. In any case, that should really be done in a separate PR.

The QMK Discord is linked in the readme (it's one of the badges) but I'll post it here too: https://discord.gg/Uq7gcHh

**Comment by GreatWizard** [comment]
the CI is going well indeed 😊 
I will definitively opening a PR only for that change

**Comment by fauxpark** [comment]
Well you've now added a core change to this PR so Travis is going to run `make all:default` instead of `make <changedkeyboards>:all`.

**Comment by GreatWizard** [comment]
I made the PR https://github.com/qmk/qmk_firmware/pull/9419
I will rebase this one when the other is merged
let's continue to talk about the "fix" over there and keep this PR related to JJ50 keymaps ^^

File: keyboards/jj50/README.md [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
+![JJ50 + SA Vilebloom (w/ Sakurios) by u/rendleshift](https://i.imgur.com/SwYZ4wol.jpg)
 
-* Keyboard Maintainer: [Wayne K. Jones](github.com/WarmCatUK)
-* Hardware Supported: JJ50 with the ATmega32a chip.
-* Hardware Availability: [KPrepublic on AliExpress](https://www.aliexpress.com/item/jj50-v1-0-Custom-Mechanical-Keyboard-50-PCB-programmed-50-preonic-layouts-bface-firmware-with-rgb/32848915277.html)
+A compact 50% (5x12) ortholinear keyboard made and sold by KPrepublic.
+
+Keyboard Maintainer: [QMK Community](https://github.com/qmk)
+Hardware Supported: Atmega32A
+Hardware Availability: [AliExpress](https://www.aliexpress.com/item/jj50-v1-0-Custom-Mechanical-Keyboard-50-PCB-programmed-50-preonic-layouts-bface-firmware-with-rgb/32848915277.html); [KPrepublic](https://kprepublic.com/collections/jj50-50/products/jj50-50-custom-keyboard-pcb-similar-with-preonic)
```
Comment (noroadsleft) [inline keyboards/jj50/README.md:9]:
```suggestion
* Keyboard Maintainer: [QMK Community](https://github.com/qmk)
* Hardware Supported: Atmega32A
* Hardware Availability: [AliExpress](https://www.aliexpress.com/item/jj50-v1-0-Custom-Mechanical-Keyboard-50-PCB-programmed-50-preonic-layouts-bface-firmware-with-rgb/32848915277.html); [KPrepublic](https://kprepublic.com/collections/jj50-50/products/jj50-50-custom-keyboard-pcb-similar-with-preonic)
```

File: keyboards/jj50/keymaps/default/README.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+## Momentarily Layers
+
+### Lower
+
+```
+,-----------------------------------------------------------------------------------.
+|   ~  |   !  |   @  |   #  |   $  |   %  |   ^  |   &  |  |   (  |   )  | Bksp |
+|------+------+------+------+------+-------------+------+------+------+------+------|
+|   ~  |   !  |   @  |   #  |   $  |   %  |   ^  |   &  |  |   (  |   )  | Del  |
```
Comment (noroadsleft) [inline keyboards/jj50/keymaps/default/README.md:61]:
```suggestion
|   ~  |   !  |   @  |   #  |   $  |   %  |   ^  |   &  |   *  |   (  |   )  | Bksp |
|------+------+------+------+------+-------------+------+------+------+------+------|
|   ~  |   !  |   @  |   #  |   $  |   %  |   ^  |   &  |   *  |   (  |   )  | Del  |
```

File: keyboards/jj50/keymaps/default/README.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+```
+,-----------------------------------------------------------------------------------.
+|   `  |   1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  | Bksp |
+|------+------+------+------+------+------+------+------+------+------+------+------|
+|   `  |   1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  | Del  |
+|------+------+------+------+------+-------------+------+------+------+------+------|
+| Del  |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |   -  |   =  |   [  |   ]  |  \   |
+|------+------+------+------+------+------|------+------+------+------+------+------|
+|      |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |ISO # |ISO / |  Up  | Down |      |
```
Comment (noroadsleft) [inline keyboards/jj50/keymaps/default/README.md:81]:
```suggestion
|      |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |ISO # |ISO / |PageUp|PageDn|      |
```

File: keyboards/jj50/keymaps/default/keymap.c [hunk 8]
```diff
@@ -19,117 +20,197 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+    /* Raise
+     * ,-----------------------------------------------------------------------------------.
+     * |   `  |   1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  | Bksp |
      * |------+------+------+------+------+------+------+------+------+------+------+------|
-     * |      |      |      |      |      |      |      |ISO ~ |ISO | |  HOM |  END |      |
+     * |   `  |   1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  | Del  |
+     * |------+------+------+------+------+-------------+------+------+------+------+------|
+     * | Del  |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |   -  |   =  |   [  |   ]  |  \   |
+     * |------+------+------+------+------+------|------+------+------+------+------+------|
+     * |      |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |ISO # |ISO / |  Up  | Down |      |
```
Comment (noroadsleft) [inline keyboards/jj50/keymaps/default/keymap.c:137]:
```suggestion
     * |      |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |ISO # |ISO / |PageUp|PageDn|      |
```

File: keyboards/jj50/config.h [hunk 9]
```diff
@@ -17,17 +18,16 @@ You should have received a copy of the GNU General Public License
...
 
-#define VENDOR_ID       0x20A0
-#define PRODUCT_ID      0x422D
+#define VENDOR_ID       0x4B50  // "KP"
+#define PRODUCT_ID      0x0050
 #define DEVICE_VER      0x0200
 #define MANUFACTURER    KPrepublic
-#define PRODUCT         JJ50
-#define DESCRIPTION     Preonic-like clone
+#define PRODUCT         KPrepublic JJ50
```
Comment (fauxpark) [inline keyboards/jj50/config.h:29]:
```suggestion
#define PRODUCT         JJ50
```

**Comment by GreatWizard** [comment]
Tested and it works like a charm :)
🎉

**Comment by aaarsene** [comment]
I justed tested it and everything seems to work like it should!

**Pull request merged**